pandas/tests/arrays/string_/test_string_arrow.py

```diff
@@ -69,6 +69,24 @@ def test_constructor_not_string_type_raises(array, chunked):
         ArrowStringArray(arr)
 
 
+@skip_if_no_pyarrow
+@pytest.mark.parametrize("chunked", [True, False])
+def test_constructor_not_string_type_value_dictionary_raises(chunked):
+    import pyarrow as pa
+
+    array = pa
+
+    arr = array.array([1, 2, 3], array.dictionary(array.int32(), array.int32()))
```

Comment:
Could you add a separate test where a dictionary type of strings does work?

Reply:
I added a case with valid input and used assert to test if value_type is string